parsing/parser.mly

```diff
@@ -115,33 +126,43 @@ let mkexp_cons consloc args loc =
 let mkpat_cons consloc args loc =
   Pat.mk ~loc (Ppat_construct(mkloc (Lident "::") consloc, Some args))
 
-let rec mktailexp nilloc = function
-    [] ->
-      let loc = { nilloc with loc_ghost = true } in
-      let nil = { txt = Lident "[]"; loc = loc } in
-      Exp.mk ~loc (Pexp_construct (nil, None))
-  | e1 :: el ->
-      let exp_el = mktailexp nilloc el in
+let mktailexp full_loc nilloc rev_list =
```

Comment:
I suggest to make `rev_list` a labelled argument. At this will be clear at the call site that we expect the list in reverse order